raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -18,117 +18,54 @@
+    public static final int DEFAULT_QUORUM_FETCH_TIMEOUT_MS = 15_000;
 
     public static final String QUORUM_ELECTION_BACKOFF_MAX_MS_CONFIG = QUORUM_PREFIX + "election.backoff.max.ms";
-    private static final String QUORUM_ELECTION_BACKOFF_MAX_MS_DOC = "Maximum time in milliseconds before starting new elections. " +
+    public static final String QUORUM_ELECTION_BACKOFF_MAX_MS_DOC = "Maximum time in milliseconds before starting new elections. " +
         "This is used in the binary exponential backoff mechanism that helps prevent gridlocked elections";
+    public static final int DEFAULT_QUORUM_ELECTION_BACKOFF_MAX_MS = 5_000;
 
     public static final String QUORUM_LINGER_MS_CONFIG = QUORUM_PREFIX + "append.linger.ms";
-    private static final String QUORUM_LINGER_MS_DOC = "The duration in milliseconds that the leader will " +
+    public static final String QUORUM_LINGER_MS_DOC = "The duration in milliseconds that the leader will " +
         "wait for writes to accumulate before flushing them to disk.";
+    public static final int DEFAULT_QUORUM_LINGER_MS = 25;
 
-    // Package-private for testing
-    static final String QUORUM_REQUEST_TIMEOUT_MS_CONFIG = QUORUM_PREFIX +
+    public static final String QUORUM_REQUEST_TIMEOUT_MS_CONFIG = QUORUM_PREFIX +
         CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG;
+    public static final String QUORUM_REQUEST_TIMEOUT_MS_DOC = CommonClientConfigs.REQUEST_TIMEOUT_MS_DOC;
+    public static final int DEFAULT_QUORUM_REQUEST_TIMEOUT_MS = 20_000;
 
-    // Package-private for testing
-    static final String QUORUM_RETRY_BACKOFF_MS_CONFIG = QUORUM_PREFIX +
+    public static final String QUORUM_RETRY_BACKOFF_MS_CONFIG = QUORUM_PREFIX +
         CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG;
-
-    static {
-        CONFIG = new ConfigDef()
-            .define(QUORUM_REQUEST_TIMEOUT_MS_CONFIG,
-                ConfigDef.Type.INT,
-                20000,
-                atLeast(0),
-                ConfigDef.Importance.MEDIUM,
-                REQUEST_TIMEOUT_MS_DOC)
-            .define(QUORUM_RETRY_BACKOFF_MS_CONFIG,
-                ConfigDef.Type.INT,
-                100,
-                atLeast(0L),
-                ConfigDef.Importance.LOW,
-                CommonClientConfigs.RETRY_BACKOFF_MS_DOC)
-            .define(QUORUM_VOTERS_CONFIG,
-                ConfigDef.Type.LIST,
-                ConfigDef.NO_DEFAULT_VALUE,
-                new ConfigDef.Validator() {
-                    @Override
-                    public void ensureValid(String name, Object value) {
-                        if (value == null) {
-                            throw new ConfigException(name, null);
-                        }
-
-                        @SuppressWarnings("unchecked")
-                        Map<Integer, InetSocketAddress> voterConnections = parseVoterConnections((List) value);
```

Comment:
We seem to have lost this validation. Can we just move it to `KafkaConfig` instead?

Reply:
See https://github.com/apache/kafka/pull/9916#discussion_r560611932